Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomDamageTarget.cs

```diff
@@ -0,0 +1,59 @@
+using System.Runtime.Versioning;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Xml.Serialization;
+using System.Xml.Schema;
+using System.Xml;
+
+public class RandomDamageTarget : ActionEffect
+{
+    public int minDamage { get; private set; }
+    public int maxDamage { get; private set; }
+
+    public RandomDamageTarget(int minDamage, int maxDamage, TargetType target)
+    {
+        this.minDamage = minDamage;
+        this.maxDamage = maxDamage;
+        Target = target;
+    }
+
+    public RandomDamageTarget()
+    {
+
+    }
+
+    public override void Apply(params CombatUnit[] targets)
+    {
+        foreach (CombatUnit target in targets)
+        {
+            int damageAmount = calculateDamageAmount(minDamage, maxDamage);
+            target.ApplyDamage(damageAmount);
+            CombatUnit attacker = CombatManager.currentCombatant;
+            DialogueBoxUIManager.AddStringToDialogueBox
+            ($"{DialogueBoxUIManager.FormatCombatUnitColor(attacker)}"
+            + $" dealt {DialogueBoxUIManager.FormatDamageColor(damageAmount)} damage to "
+            + $"{DialogueBoxUIManager.FormatCombatUnitColor(target)}");
+        }
+    }
+
+    private int calculateDamageAmount(int minDamage, int maxDamage)
+    {
+        System.Random randomNum = new System.Random();
```

Comment:
Is there a reason for using System.Random rather than UnityEngine.Random?

Reply:
This seems to be the main difference : "UnityEngine.Random is a static class, so we will have just one generator of random numbers per match.With System.Random you have to instantiate at least one, or you can instantiate multiple Random generators with different seeds at the same time". From reading a bit more it seems like UnityEngine.Random might be better for us

Reply:
Yeah you're exactly right. System.Random is fine, but it seems like just using UnityEngine.Random would eliminate the need to create and seed a random number generator in our code.